neuraxle/pipeline.py

```diff
@@ -205,6 +271,7 @@ def _transform_core(self, data_container: DataContainer) -> DataContainer:
         steps_left_to_do, data_container = self._load_pipeline_checkpoint(data_container)
 
         for step_name, step in steps_left_to_do:
+            step.set_parent(self)
```

Comment:
Is this a code smell? Should `self` be contained in the `data_container` such that it isn't setted at each iteration again into the data_container? 

If so, we might need renaming the data_container as something else that is more like an execution context.

Reply:
It might be cool to have the current path of the step to be contained in the data_container: this way, it would be easily passed and the step would know where to save itself. I don't remember when the path is passed to a sub step in a pipeline, so I might be saying crap. 

Would you be against what I hereby suggest? Could you please find arguments against doing that?

Reply:
I had that idea 'Should self be contained in the data_container such that it isn't setted at each iteration again into the data_container?'... I almost chose that solution, but I am still unsure. to be discussed.